pkg/updater/download.go

```diff
@@ -148,7 +153,8 @@ func extractTarGzFile(targetPath string, reader io.Reader) error {
 		return fmt.Errorf("could not create file: %w", err)
 	}
 	defer f.Close()
-	_, err = io.Copy(f, reader)
+	limitedReader := &io.LimitedReader{R: reader, N: maxFileSize}
```

Comment:
I assume that we should also limit at the entry's header size too.

As a sidenote, I think the limitedreader won't reject the read but just read up to the limit - not sure if we want a hard error in those cases.

Reply:
Controlling the read size will require a complete drain on the reader, and by definition, it will load the memory, which is what we want to avoid.
I changed the control to keep streaming and enforce an upper filesize limit written on disk.